### PR TITLE
Add YouTube Features

### PR DESCRIPTION
**Pull request by kuhasuki**
- Add a link to your stream in live dock
- Add a link to live dashboard in live dock
- Allow user to change title and description for a YouTube stream in progress
- Fetch donation_url for user and attach to description when enabled by user
- Log users into YouTube chat automatically

### CODE REVIEW

File: app/components/LiveDock.vue [hunk 1]
```diff
@@ -48,8 +48,12 @@
       </div>
 
       <div class="live-dock-info">
-        <a @click="showEditStreamInfo" v-if="isTwitch">Edit Stream Info</a>
-        <a @click="refreshChat">Refresh Chat</a>
+        <div class="platform-tools">
```
Comment (andycreeth) [inline app/components/LiveDock.vue]: Let's make this class `live-dock-platform-tools` just to match everything else in this component and ensure our class names are sufficiently unique.

File: app/services/platforms/youtube.ts [hunk 2]
```diff
@@ -57,6 +57,13 @@ export class YoutubeService extends Service implements IPlatformService {
     });
   }
 
+  fetchDescriptionString(): Promise<string> {
```
Comment (andycreeth) [inline app/services/platforms/youtube.ts]: I would call this `fetchDescription`.  The nice thing about typescript is you already know it returns a string.

File: app/services/platforms/youtube.ts [hunk 3]
```diff
@@ -150,25 +157,28 @@ export class YoutubeService extends Service implements IPlatformService {
...
-        headers,
-        body: JSON.stringify(data)
-      }
-    );
+  putChannelInfo(streamTitle: string, streamDescription: string): Promise<boolean> {
+    return this.fetchDescriptionString().then(autopublishString => {
+      const headers = new Headers();
+      headers.append('Content-Type', 'application/json');
+
+      const fullDescription = autopublishString.concat(streamDescription);
```
Comment (andycreeth) [inline app/services/platforms/youtube.ts]: Is a simple concatenation enough?  Should we be adding a newline or something?  Not really sure.

Comment (kuhasuki) [reply app/services/platforms/youtube.ts]: The autopublish string has a newline in it if enabled.

File: app/services/platforms/youtube.ts (hunk 2 above)
Comment (kuhasuki) [reply app/services/platforms/youtube.ts]: Good Point

File: app/components/LiveDock.vue (hunk 1 above)
Comment (kuhasuki) [reply app/components/LiveDock.vue]: fixed 3d9cce1c30c4e8096068868317ba472e91a5cf4a

File: app/components/Chat.vue.ts [hunk 4]
```diff
@@ -29,7 +30,18 @@ export default class Chat extends Vue {
     const webview = this.$refs.chat;
     const settings = this.customizationService.getSettings();
 
-    service.getChatUrl(nightMode).then(chatUrl => this.chatUrl = chatUrl);
+    if (service instanceof YoutubeService) {
+      service.getChatUrl(nightMode).then(chatUrl => {
+        this.chatUrl = 'https://youtube.com/signin';
```
Comment (andycreeth) [inline app/components/Chat.vue.ts]: If the user is already logged in, this should immediately redirect, right?

Comment (kuhasuki) [reply app/components/Chat.vue.ts]: Yup

**Pull request merged**